### PR TITLE
Implement check workflow

### LINKED ISSUE #572: Implement new License Check
**Describe the feature request**
As a plugin maintainer I want to check if my dependencies are still as checked.

**Upstream changes**
* [x] create GitHub Workflow for license information diff 
* [x] Integrate License Check Badge

**Additional context**

### PR DESCRIPTION
**Pull request by MartinGroscheTT**
Resolves #572

### CODE REVIEW

File: config/check_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+    # check for correct schema first
+    try:
+        jsonschema.validate(allow_json, schema_json)
+    except jsonschema.exceptions.ValidationError as error:
+        found_errors.append(f"allowlist does not comply to the schema...\n{error.message}")
+
+    # dependencies
+    allowed_packages = [item["moduleName"] for item in allow_json["allowedLicenses"]]
+    sbom_packages = ["{}:{}".format(item["group"], item["name"]) for item in sbom_json["components"]]
```
Comment (ErikRehmTT) [inline config/check_dependencies.py:100]: Wouldn't it be easier and less complex, to transform them once into a dict each, with the module name as key. Here you could create warning, when a component is listed twice. 

For the diff you could use the .keys()

and for the actual comparison you would iterate through one dict and get the matching entry in the other one by simple accessing it

File: config/check_dependencies.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        self.__name()
+        self.__licenses()
+        if not product:
+            self.__version()
+        return self.errors
+
+    def __name(self):
+        sbom_name = "{}:{}".format(self.sbom["group"], self.sbom["name"])
+        if self.allow["moduleName"] != sbom_name:
+            self.append_error("name", self.allow["moduleName"], sbom_name)
```
Comment (ErikRehmTT) [inline config/check_dependencies.py:55]: ComponentValidator is only instanciated, when the names match, so this check will be fine all the time

**Review by ErikRehmTT — CHANGES_REQUESTED** [review]

File: config/check_dependencies.py (hunk 1 above)
Comment (MartinGroscheTT) [reply config/check_dependencies.py:100]: The list generation is more complex but afterwards it only a string comparison

Comment (ErikRehmTT) [reply config/check_dependencies.py:100]: But 
```
for component in sbom_json["components"]:
        # component["license"] = metadata(component["name"])["license"]
        for allowed_component in allow_json["allowedLicenses"]:

            name = allowed_component["moduleName"] == "{}:{}".format(component["group"], component["name"])
            if name:
                validator = ComponentValidator(allowed_component, component)
                ...
```

is way more complex than 

```
for name, sbom_entry in sbom_packages.items():
        if name in allow_packages:
                validator = ComponentValidator(allow_packages[name], sbom_entry)
                ...
```

Comment (MartinGroscheTT) [reply config/check_dependencies.py:110]: Keys() are unique. So we have to check multiple modules the other way

**Review by ErikRehmTT — APPROVED** [review]

**Pull request merged**